cmd/kube-controller-manager/app/core.go

```diff
@@ -581,6 +581,23 @@ func startTTLAfterFinishedController(ctx context.Context, controllerContext Cont
 	return nil, true, nil
 }
 
+func startLegacySATokenCleaner(ctx context.Context, controllerContext ControllerContext) (controller.Interface, bool, error) {
```

Comment:
Hi Stanislav, I am sorry but I am not sure I get the meaning. Would you please clarify a little bit? This startLegacySATokenCleaner is called by `register("legacy-service-account-token-cleaner", startLegacySATokenCleaner)` in the NewControllerInitializers function in k8s.io/kubernetes/cmd/kube-controller-manager/app/controllermanager.go. It is the same place where the startServiceAccountController function is called.

Reply:
What I mean it to simply move the function from `app/core.go` to `app/controllermanager.go` near the SA tokens controller so that these similar controllers are somewhat grouped together.

Reply:
I see. So you mean I need to put this controller close to the SA token controller, not the SA controller. Now all the controllers except the SA token controller are registered in the `NewControllerInitializers` function in app/controllermanager.go and the starting functions for these controllers are put in app/core.go. The SA token controller is special and needs to be started firstly to set up permissions for other controllers . IMHO, we can just leave this token cleaner together with other controllers (which include the SA controller).

Reply:
I agree that this is a reasonable place to put this, and that the token controller is special in ways that this controller is not